frontends/p4/fromv1.0/programStructure.cpp

```diff
@@ -1752,8 +1760,16 @@ ProgramStructure::convertAction(const IR::ActionFunction* action, cstring newNam
 
     for (auto p : action->action) {
         auto stat = convertPrimitive(p);
-        if (stat != nullptr)
-            body->push_back(stat); }
+        if (stat != nullptr) {
+            body->push_back(stat); 
+            if (auto mcs = stat->to<IR::MethodCallStatement>()) {
```

Comment:
Why not construct `annots` using the copy constructor that takes annotations. I.e., `annots = new IR::Annotations(mcs->getSourceInfo(), mcs->getAnnotations()->annotations)` and directly add to the body?
Essentially, this whole code can be:
```
if (stats != nullptr) {
    body->push_back(stat);
    if (auto mcs = stat->to<IR::MethodCallStatement>())
        body->annotations = new IR::Annotations(mcs->getSourceInfo(), mcs->getAnnotations()->annotations);
}
```